wordpress-mobile/WordPress-iOS
# Migrates Reader remotes + models + tests into WordPressKit

Fixes #7323 
Fixes #7324 
Fixes #7325 

**Models:**
* Migrates the `RemoteReaderPost.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderSite.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderSiteInfo.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderTopic.h/.m` model to the WPKit framework
* Migrates the `RemoteSourcePostAttribution.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderCrossPostMeta.swift` model to the WPKit framework

**Remotes:**
* Migrates the `ReaderPostServiceRemote.h/.m` and related tests into the WPKit framework
* Migrates the `ReaderSiteServiceRemote.h/.m` and related tests into the WPKit framework
* Migrates the `ReaderTopicServiceRemote.h/.m` and related tests into the WPKit framework

**WPShared:**
* Moves `BasePost`'s `summaryFromContent` class method to`NSString+Helpers.h/.m` in WPShared (including static const `PostDerivedSummaryLength`)
* Moves `testSummaryForContent` test to WPShared's `NSStringHelpersTests.m` (and deletes `BasePostTest.m` because that was the only unit test in that file)
* Migrates `PhotonImageURLHelper.h/.m` and related tests into the WPShared framework
* Migrates `DateUtils.h/.m` to the WPShared framework
* Migrates `RichContentFormatter.swift` and related tests into the WPShared framework (purposefully left 2 tests - `RichContentFormatterTests.swift` - in main WordPress app because of dependency on `UIApplication.shared.keyWindow`)
* Migrates `WPImageURLHelper.swift' to the WPShared framework

**Misc:**
* Updates related imports/refs in main WordPress project
* Migrates missed tests (NotificationSyncServiceRemoteTests.swift) from previous notes PR to WPKit

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. Exercise the reader thoroughly. Try to test every aspect of its feature set.

@aerych given your supreme reader expertise, would you mind reviewing? _Greatly_ appreciated!
